src/java.base/share/classes/javax/crypto/Cipher.java

```diff
@@ -2592,6 +2592,41 @@ public final byte[] wrap(Key key)
         return spi.engineWrap(key);
     }
 
+    /**
+     * Export a derived key based on the current cryptographic state and
+     * additional context.
+     *
+     * <p>This method is designed to enable the generation of additional
+     * keys for use in various cryptographic operations, ensuring that key
+     * material can be securely derived from the existing encryption or
+     * decryption state.
+     *
+     * <p>This method guarantees that an encryption cipher and a decryption
+     * cipher, if initialized with the same symmetric key or a matching
+     * asymmetric key pair and equivalent parameters, will produce identical
+     * derived key when the same arguments are provided.
+     *
+     * @param context a byte array representing additional data or context
+     *          information that influences the key derivation process.
+     *          The derived key should be unique to the given context.
+     * @param algorithm the algorithm of the derived key
+     * @param length the desired length of the derived key in bytes
+     *
+     * @return the derived key
+     *
+     * @throws UnsupportedOperationException if the corresponding method in the
+     *          {@code CipherSpi} is not supported
+     * @throws IllegalArgumentException if one of the input arguments is invalid
+     * @throws IllegalStateException if this {@code Cipher} object is in a wrong
+     *          state (e.g., has not been initialized)
+     *
+     * @since 23
+     */
+    public final SecretKey exportKey(byte[] context, String algorithm, int length) {
```

Comment:
You've probably thought about this already but adding a final method to a non-final class with an accessible constructor is source and binary incompatible.

Reply:
Thanks. I'll change it to non-final. I just hope nobody uses the same name with same arguments but different return type in their subclasses. Seems not likely.